### PR TITLE
Issue #11508: Added LITERAL_YIELD token support to WhitespaceAfter

### LINKED ISSUE #11508: WhitespaceAfter should support LITERAL_YIELD token
https://checkstyle.org/config_whitespace.html#WhitespaceAfter

I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
/var/tmp $ javac WithYield.java
/var/tmp $

/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAfter">
            <property name="tokens" value="LITERAL_YIELD"/>
        </module>
    </module>
</module>

/var/tmp $ cat WithYield.java
public class WithYield {
    public static void main(String[] args) {
        int a = switch (args[0]) {
            case "got":
                yield 1;
            case "my":
                yield(3);
            default:
                yield 2;
        };
        System.out.println(a);
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.1-all.jar -c config.xml WithYield.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - Token "LITERAL_YIELD" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:476)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:403)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:330)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:189)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Token "LITERAL_YIELD" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.TreeWalker.registerCheck(TreeWalker.java:223)
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:133)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:471)
        ... 5 more
Checkstyle ends with 1 errors.
```

**Is your feature request related to a problem? Please describe.**
It would be nice if checkstyle could detect missing space after `yield` keyword.

**Describe the solution you'd like**
WhitespaceAfter could support new literal type - `LITERAL_YIELD`

#11489, #11493 and #11494 are similar requests for `try`/`finally`/`synchronized` keywords support

### PR DESCRIPTION
**Pull request by Binita-tech**
Resolves #11508 
Diff Regression config: https://gist.githubusercontent.com/Binita-tech/76be87d6c361ab5305863361aef3de15/raw/b207f4d59a809e58e13b45863e94479b7b3226e2/config.xml
Diff Regression patch config: https://gist.githubusercontent.com/Binita-tech/6fbe5f120efde70be0ad345079f86acc/raw/ef0b6c9270a7f0c03d82c67ce4ef9f2ed19be38f/patch_config.xml

### CODE REVIEW

**Comment by Binita-tech** [comment]
I have made input file in resources-noncompilable folder. When I am putting this input file name in test, I am getting error of NoSuchFileException. Please guide on how to write such input file name in tests.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java [hunk 1]
```diff
@@ -162,6 +162,16 @@ public void testLiteralDo() throws Exception {
                 expected);
     }
 
+    @Test
+    public void testLiteralYield() throws Exception {
+        final String[] expected = {
+            "16:9: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "yield"),
+        };
+        verifyWithInlineConfigParser(
+                getPath("InputWhitespaceAfterLiteralYield.java"),
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java:171]: There is method getnoncompiledpath is something like this.
Please use it

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java (hunk 1 above)
Comment (Binita-tech) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java]: Done

**Comment by Binita-tech** [comment]
GitHub, generate report

**Comment by Binita-tech** [comment]
GitHub, generate report

**Comment by romani** [comment]
Please add this to inputs https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/a9e5397_2022201703/reports/diff/openjdk17/index.html#A1

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralYield.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+WhitespaceAfter
+tokens = LITERAL_YIELD
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.whitespaceafter;
```
Comment (romani) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralYield.java:8]: You need comment like 
https://github.com/checkstyle/checkstyle/blob/a9e53970c2809a408ce05f24c7c1a380923a8f11/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterSwitchStatements.java#L7

To let CI pass

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralYield.java (hunk 2 above)
Comment (Binita-tech) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralYield.java:9]: Done

**Comment by romani** [comment]
Please compile Input on your local https://cloud.drone.io/checkstyle/checkstyle/10221/5/8

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java [hunk 3]
```diff
@@ -110,8 +112,15 @@
...
  *
+ *      synchronized (this) { } // OK
  *      synchronized(this) { } // violation ''synchronized' is not followed by whitespace'
- *      synchronized (this) { } // ok
+ *
+ *      int a = switch (args[0]) {
+ *        case "got":
+ *          yield (1); // OK
+ *        case "my":
+ *          yield(3); // violation ''yield' is not followed by whitespace'
```
[the remark this answers is not in the document]
Comment (Binita-tech) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Done

**Review by romani — APPROVED** [review]
Ok to merge

**Review by pbludov — APPROVED** [review]

**Comment by pbludov** [comment]
GitHub, generate site

**Pull request merged**